### PR TITLE
Add Optimas

### PR DESCRIPTION
**Pull request by ax3l**
Optimas is a Python library designed for highly scalable optimization, from laptops to massively-parallel supercomputers.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] ~Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ax3l** [comment]
@conda-forge/help-python ready for review :)

**Comment by AngelFP** [comment]
I agree to be listed in this packages as a co-maintainer.

**Comment by RemiLehe** [comment]
I agree to be listed in this packages as a co-maintainer.

**Review by AngelFP — COMMENTED** [review]
Suggested changing python version to >3.9, since this is what optimas currently requires.

Should we also set a minimum requirement for Ax and libensemble as in https://github.com/optimas-org/optimas/blob/main/pyproject.toml?

**Comment by ax3l** [comment]
> Should we also set a minimum requirement for Ax and libensemble as in https://github.com/optimas-org/optimas/blob/main/pyproject.toml?

These minimal requirements are not really needed, since conda-forge "lives at the latest sphipped release" by default. in fact, too many constraints often confuse its concretizer :sweat: The Python one is just required for `noarch: python` packages.

File: recipes/optimas/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{% set version = "0.3.1" %}
+
+package:
+  name: optimas
+  version: {{ version }}
+
+source:
+  fn: {{ version }}.tar.gz
```
Comment (ax3l) [inline recipes/optimas/meta.yaml:8]:
```suggestion
```

File: recipes/optimas/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  url: https://github.com/optimas-org/optimas/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 875ffe6e791f167b96edda1636c5ab10e3a863e961ca8f4a41d1e5c5aa559112
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (ax3l) [inline recipes/optimas/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

**Pull request merged**